### PR TITLE
Check Marketplaces Match Server Type

### PR DESCRIPTION
**Pull request by BEAdi**
## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8938)
relates: https://github.com/demisto/content/pull/31257

## Description
Add support for running tests on a specific marketplace.

### CODE REVIEW

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 1]
```diff
@@ -439,12 +476,59 @@ def test_has_skipped_integration():
 
             return False
 
+        def marketplaces_match_server_type():
+            """
+            Checks if the test has a marketplace value, and if so- if it suits the server machine we are on.
```
Comment (dorschw) [inline demisto_sdk/commands/test_content/TestContentClasses.py:481]: please document a bit more on the return type and why it matters, consider changing the name to `does_marketplace_match_server_type`

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 2]
```diff
@@ -439,12 +476,59 @@ def test_has_skipped_integration():
...
+                )
+            self.log_info(
+                f"MARKETPLACES FOR TEST PLAYBOOK:\n"
+                f"{self.configuration.marketplaces=} ({test_server_types=})\n"
+                f"for {self.configuration.playbook_id=}\n"
+                f"with {self.configuration.test_instance_names=}\n"
+                f"in {self.build_context.server_type=}"
+            )
+            if not test_server_types:
+                return True
```
Comment (dorschw) [inline demisto_sdk/commands/test_content/TestContentClasses.py:496]: please explain with just a few words, on each `return`, why we return it - it's a bit unclear

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 3]
```diff
@@ -52,17 +53,27 @@
...
 MAX_RETRIES = 3
 RETRIES_THRESHOLD = ceil(MAX_RETRIES / 2)
 
 SLACK_MEM_CHANNEL_ID = "CM55V7J8K"
+XSOAR_SERVER_TYPE = "XSOAR"
 XSIAM_SERVER_TYPE = "XSIAM"
+XPANSE_SERVER_TYPE = "XPANSE"
 XSOAR_SAAS_SERVER_TYPE = "XSOAR SAAS"
 
+MARKETPLACE_VERSIONS_TO_SERVER_TYPE = {
```
Comment (dorschw) [inline demisto_sdk/commands/test_content/TestContentClasses.py:69]: Please make sure this is tested thoroughly

**Review by dorschw — COMMENTED** [review]
🚀

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**